fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1984,6 +1984,34 @@ void luceneCursorTestAllMatchesSkipped() throws Exception {
         }
     }
 
+    @Test
+    void forceMerge() {
```

Comment:
I'm not sure how we'd right dependable unit tests here, but it seems like having tests that show that regular reads, writes, and updates can all happen while the first thread is doing a force merge.

Maybe there needs to be a way to that it to only begin to bulk merge, or something, to write tests.

Reply:
Perhaps it can hook `openLikeContext` to do another transaction in between those of the directory.

Reply:
`IndexWriter` expects that a lock will prevent another writer, normally in the file system, on the same directory.

Reply:
* https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexWriter.java#L948 acquire lock.
* https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexWriter.java#L1151-L1163 delete any "orphan" files.

Since our locks are noops, it can delete a file that is part of a parallel merge.